File: ZhigaoWeather/app/src/main/java/com/weather/zhigao/LogUtil.java
/*
 * File Name: LogUtils.java
 * History:
 * Created by lipan on 2014-4-4
 */
package com.weather.zhigao;

import android.util.Log;


/**
 * 日志输出控制类 (Description)
 *
 * @author liyusheng
 */
public class LogUtil {
    public static boolean isDebug = BuildConfig.DEBUG;
    public static boolean isNetDebug = true;
    private static final String TAG = "gaolei";


    public static void d(String msg) {
        if (isDebug)
            Log.d(TAG, msg);
    }

    public static void d(String tag, String msg) {
        if (isDebug)
            Log.d(tag, msg);
    }

    public static void nd(String tag, String msg) {
        if (isDebug && isNetDebug)
            Log.d(tag, msg);
    }

    public static void v(String msg) {
        if (isDebug)
            Log.v(TAG, msg);

    }

    public static void e(String msg) {
        if (isDebug)
            Log.e(TAG, msg);
    }

    public static void e(Exception e) {
        if (isDebug)
            Log.e(TAG, e.getMessage().toString());
    }
}

File: ZhigaoWeather/app/src/main/java/com/weather/zhigao/utils/LogUtil.kt
/*
 * File Name: LogUtils.java
 * History:
 * Created by lipan on 2014-4-4
 */
package com.weather.zhigao.utils

import android.util.Log

import com.weather.zhigao.BuildConfig


/**
 * 日志输出控制类 (Description)
 *
 * @author liyusheng
 */
object LogUtil {
    var isDebug = BuildConfig.DEBUG
    var isNetDebug = true
    private val TAG = "gaolei"


    fun d(msg: String) {
        if (isDebug)
            Log.d(TAG, msg)
    }

    fun d(tag: String, msg: String) {
        if (isDebug)
            Log.d(tag, msg)
    }

    fun nd(tag: String, msg: String) {
        if (isDebug && isNetDebug)
            Log.d(tag, msg)
    }

    fun v(msg: String) {
        if (isDebug)
            Log.v(TAG, msg)

    }

    fun e(msg: String) {
        if (isDebug)
            Log.e(TAG, msg)
    }

    fun e(e: Exception) {
        if (isDebug)
            Log.e(TAG, e.message.toString())
    }
}
